lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -0,0 +1,397 @@
+                ips=[self._get_core_network_ip_config()],
+            ),
+        ]
+```
+"""
+
+import json
+import logging
+from dataclasses import asdict, dataclass
+from json.decoder import JSONDecodeError
+from typing import Callable, Optional
+
+import httpx
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.generic_resource import GenericNamespacedResource, create_namespaced_resource
+from lightkube.models.apps_v1 import StatefulSetSpec
+from lightkube.models.core_v1 import (
+    Capabilities,
+    Container,
+    PodSpec,
+    PodTemplateSpec,
+    SecurityContext,
+)
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.resources.apps_v1 import StatefulSet
+from lightkube.types import PatchType
+from ops.charm import CharmBase, EventBase, RemoveEvent
+from ops.framework import Object
+
+# The unique Charmhub library identifier, never change it
+LIBID = "75283550e3474e7b8b5b7724d345e3c2"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+NetworkAttachmentDefinition = create_namespaced_resource(
+    group="k8s.cni.cncf.io",
+    version="v1",
+    kind="NetworkAttachmentDefinition",
+    plural="network-attachment-definitions",
+)
+
+
+@dataclass
+class NetworkAnnotation:
+    """NetworkAnnotation."""
+
+    name: str
+    interface: str
+    ips: Optional[list] = None
+
+    dict = asdict
+
+
+class KubernetesMultusError(Exception):
+    """KubernetesMultusError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class KubernetesClient:
+    """Class containing all the Kubernetes specific calls."""
+
+    def __init__(self, namespace: str):
+        self.client = Client()
+        self.namespace = namespace
+
+    def network_attachment_definition_is_created(self, name: str) -> bool:
+        """Returns whether a NetworkAttachmentDefinition is created.
+
+        Args:
+            name: NetworkAttachmentDefinition name
+
+        Returns:
+            bool: Whether the NetworkAttachmentDefinition is created
+        """
+        try:
+            self.client.get(
+                res=NetworkAttachmentDefinition,
+                name=name,
+                namespace=self.namespace,
+            )
+            logger.info(f"NetworkAttachmentDefinition {name} already created")
+            return True
+        except ApiError as e:
+            if e.status.reason == "NotFound":
+                logger.info(f"NetworkAttachmentDefinition {name} not yet created")
+                return False
+            else:
+                raise KubernetesMultusError(
+                    f"Unexpected outcome when retrieving network attachment definition {name}"
+                )
```

Comment:
We can remove unnecessary nesting.
```suggestion
           if e.status.reason != "NotFound":
                raise KubernetesMultusError(
                    f"Unexpected outcome when retrieving network attachment definition {name}"
                )
            logger.info(f"NetworkAttachmentDefinition {name} not yet created")
            return False
```